Research/Research/RSDFont+Lato.swift

```diff
@@ -0,0 +1,169 @@
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+#if os(macOS)
+import AppKit
+#else
+import UIKit
+#endif
+
+extension RSDFont {
+    
+    /// Returns the most appropriate Lato font found by the system. This will first check for the
+    /// Lato font that matches the requested weight. If not found (because this framework only
+    /// includes a subset of the full Lato font family), then it will fall back to the most
+    /// appropriate font registered from this framework.
+    ///
+    /// This framework does not include all the possible sizes and weights for the Lato font
+    /// because the full font family is about 11 mb.
+    ///
+    /// - parameters:
+    ///     - fontSize: The size of the font requested.
+    ///     - weight: The weight of the font requested.
+    /// - returns: The closest font to the one requested that is registered for this application.
+    public static func latoFont(ofSize fontSize: CGFloat, weight: RSDFont.Weight = .regular) -> RSDFont {
+        let fontName = LatoFontWrapper.shared.fontName(weight)
+        let registeredFont = RSDFont(name: fontName.preferred, size: fontSize) ??
+            RSDFont(name: fontName.fallback, size: fontSize)
+        guard let font = registeredFont else {
+            print("WARNING! Failed to return font `\(fontName)`. Using system font.")
+            return RSDFont.systemFont(ofSize: fontSize, weight: weight)
+        }
+        return font
+    }
+    
+    /// Returns the Lato italic font embedded in this framework.
+    /// - seealso: `latoFont()`
+    public static func italicLatoFont(ofSize fontSize: CGFloat, weight: RSDFont.Weight = .regular) -> RSDFont {
+        let fontName = LatoFontWrapper.shared.italicFontName(weight)
+        let registeredFont = RSDFont(name: fontName.preferred, size: fontSize) ??
+            RSDFont(name: fontName.fallback, size: fontSize)
+        guard let font = registeredFont else {
+            print("WARNING! Failed to return font `\(fontName)`. Using system font.")
+            return RSDFont.italicSystemFont(ofSize: fontSize)
```

Comment:
Should it also try to use the requested font weight? Or is there not a way to do that here?

Reply:
No, can't set the weight on the system font using this method.